### PR TITLE
[WIP] Simplify engine exports.

### PR DESCRIPTION
**Pull request by rwjblue**
- Avoid relying on initializer to setup `.reopen`'s needed.
- Use `.extend()` when possible (instead of `Engine.reopen` we can
  `.extend()`)
- Remove initializer (it is not needed internally).
- Remove all but one usage of `Ember.__loader.require` (remaining usage
  is for `hasDefaultSerializer`).

### CODE REVIEW

File: tests/unit/initializers/engines-test.js [hunk 1]
```diff
@@ -15,8 +14,6 @@ module('Unit | Initializer | engines', {
 
 // Replace this with your real tests.
 test('it works', function(assert) {
-  EnginesInitializer.initialize(application);
-
   // you would normally confirm the results of the initializer here
   assert.ok(true);
 });
```
Comment (nickiaconis) [inline tests/unit/initializers/engines-test.js]: This test file doesn't do anything productive anymore. Do you have plans for it or should it be removed?

Comment (rwjblue) [inline tests/unit/initializers/engines-test.js]: good catch, will remove

**Comment by trentmwillis** [comment]
I tested this out locally, and it breaks the lazy loading runtime behavior, since the Router extension doesn't occur until after the Router instance has been created.

I'm not entirely sure why the tests don't break, probably has to do with preloading the assets.

**Comment by scalvert** [comment]
@rwjblue any thoughts on how to resolve this?

**Comment by rwjblue** [comment]
@scalvert - Yes, just lack of time to complete.  Ping me in slack if you would like to pick it up.

**Comment by scalvert** [comment]
Will do

**Pull request closed without merging**